kubespawner/reflector.py

```diff
@@ -228,7 +228,18 @@ async def _list_and_update(self, resource_version=None):
             kwargs["namespace"] = self.namespace
 
         list_method = getattr(self.api, self.list_method_name)
+
         initial_resources_raw = await list_method(**kwargs)
+        if not initial_resources_raw.ok:
+            self.log.error(
```

Comment:
Can you use `self.log.exception` here instead, and pass it the exception you're creating?

Reply:
Yes, but as far as I know, it's not possible without a try-catch. I've pushed a fix

Reply:
Based on [python docs](https://docs.python.org/3/library/logging.html#logging.Logger.exception):

> This method should only be called from an exception handler.

So I think you were forced to do that for this to make sense. But, doesn't this mean we get stack trace details from raising the error a few lines above, and that in turn doesn't help us get informed about the original error anyhow associated with getting a response where `response.ok == False`?

Reply:
With that in mind, was the original thing better or not? I'm not sure - I'll defer to @yuvipanda to make a decision and then we go for it.